SwissDataScienceCenter/renku-ui
# feat(client): support session autosaves

This PR implements most of the changes from #1345 , except the visual changes to the commits info in the dropdown menu.
There are a few additional changes to simplify selecting an already running session for a target branch.

In short, these are the changes:
* Get and delete autosaves.
* Select new session commit based on autosaves and running sessions.
* Update CheckNotebookStatus component to support per-branch sessions.

**How to test**
Start a session on any project (possibly from last - 1 or -2 commit), do some work, and stop the session.
You should now have an autosave. If you try to start a session again, you should see some info when trying to select a branch/commit (P.S. this is silent unless you click on "select branch/commit/..." -- see screenshots below).
Notice that the older commits should be selected when
- No session is running, but the autosave is detected.
- The session is already running (take priority over the last commit)
- File preview on notebooks should also detect running sessions form older commits

**Screenshots**
* Autosave content available
![Screenshot_20220117_154255](https://user-images.githubusercontent.com/43481553/149918903-90e4f789-3c97-4377-a7bc-e670fac71681.png)

* User trying to start from another commit
![Screenshot_20220117_154313](https://user-images.githubusercontent.com/43481553/149918960-19487279-8cd5-44b2-b510-8d8cf9e481fc.png)

* Error on delete --> we don't get any info about the error, but it may (rarely) happen if the user has already "consumed" the autosave in another tab and left this open 
![Screenshot_20220117_134806](https://user-images.githubusercontent.com/43481553/149919016-9ab1dba1-8580-46a8-a999-d19cd8c9a99f.png)



/deploy renku=1.0-next renku-graph=cli-v1 renku-core=0000-disbale-check-migration-optimization
fix #1345, #789
